.husky/commit-msg

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npx --no -- commitlint --edit $1
```

Comment:
I am trying to find what does --no but all I can find it --no-install and I can't find --no can you please help me @mikejpeters

Reply:
@mikejpeters  I also did:

i executed `npm i` and `npm run prepare` 
I then proceeded to edit the readme to make silly commit messages like these:

```
commit a365c96db1d6199f3993f5c022274857b688d66a (HEAD -> misc-chores)
Author: Shereef Marzouk <shereef.marzouk@daysmart.com>
Date:   Sun Feb 27 13:22:05 2022 -0400

    Tuh4iuyuifkjsdjklswkj 3

commit 789e487e68d691cd647c83e6fae6ee1e47e446e9
Author: Shereef Marzouk <shereef.marzouk@daysmart.com>
Date:   Sun Feb 27 13:21:41 2022 -0400

    KHGDJHGSD: kjashkjhgaskjd

commit c43cd57b9d531113074a5fc583b93661137640ab
Author: Shereef Marzouk <shereef.marzouk@daysmart.com>
Date:   Sun Feb 27 13:21:21 2022 -0400

    : ascasdsad

commit 376a58dfe4a3b9630f86c6f1c459c01ffe365e8b (origin/misc-chores)
Author: Mike Peters <mikejpeters@gmail.com>
Date:   Sat Feb 26 16:05:10 2022 -0800

    chore: automate release process
    
    closes #146
```

it committed fine 
What am I doing wrong ?

I feel like at least 2 of these go against `@commitlint/config-conventional`

I haven't used semantic-release, commitlint or commitizen before but I am looking forward to learning them

Reply:
By default if you try to run a package with `npx` that isn't installed, it will prompt you to proceed:

```
$ npx commitlint
Need to install the following packages:
  commitlint
Ok to proceed? (y)
```

But in the `commit-msg` hook, prompts wouldn't work correctly, so it's better to suppress the prompt (which can be done with `--yes` or `--no`). Here we prefer `--no` since the package should be installed with the rest of the dev dependencies, and if it isn't, better to get an error right away.

From [npx](https://docs.npmjs.com/cli/v8/commands/npx) docs:

> If any requested packages are not present in the local project dependencies, then they are installed to a folder in the npm cache, which is added to the PATH environment variable in the executed process. A prompt is printed (which can be suppressed by providing either --yes or --no).

Reply:
Hm I'm not sure, can you please share the output from `git commit`?

Reply:
```
[2022-02-27T17:39:19.659Z] > git add -A -- . [70ms]
[2022-02-27T17:39:19.706Z] > git -c user.useConfigOnly=true commit --quiet --allow-empty-message --file - [46ms]
[2022-02-27T17:39:19.706Z] hint: The '.husky/pre-commit' hook was ignored because it's not set as executable.
hint: You can disable this warning with `git config advice.ignoredHook false`.
hint: The '.husky/commit-msg' hook was ignored because it's not set as executable.
hint: You can disable this warning with `git config advice.ignoredHook false`.
[2022-02-27T17:39:19.751Z] > git symbolic-ref --short HEAD [19ms]
[2022-02-27T17:39:19.776Z] > git for-each-ref --format=%(refname)%00%(upstream:short)%00%(objectname)%00%(upstream:track) refs/heads/misc-chores refs/remotes/misc-chores [25ms]
[2022-02-27T17:39:19.802Z] > git remote --verbose [25ms]
[2022-02-27T17:39:19.901Z] > git for-each-ref --sort -committerdate --format %(refname) %(objectname) %(*objectname) [124ms]
[2022-02-27T17:39:19.922Z] > git config --get commit.template [20ms]
[2022-02-27T17:39:19.941Z] > git config --get commit.template [17ms]
[2022-02-27T17:39:20.941Z] > git ls-files --stage -- /Users/shereef.marzouk/Dev/serverless-finch/README.md [65ms]
[2022-02-27T17:39:20.983Z] > git cat-file -s 79b4ecde7f27fe785fcbd2d561dae0d7697c1fb1 [40ms]
```
this is my VSCode output in the git tab


I see 
hint: The '.husky/pre-commit' hook was ignored because it's not set as executable.
hint: You can disable this warning with `git config advice.ignoredHook false`.

Reply:
I messed up and merged this :( But I've fixed the hook permissions on master. Can you pull the latest version and confirm that the hooks run for you?